### ISSUE TITLE
How to use segmentation labels when trying to train a new registration model by using train_synth_reg.py?

### ISSUE THREAD

**Issue by MoXiaoYing96** [voxelmorph/voxelmorph#288]
Hi, I'am trying to train a new registration model by using **_train_synth_reg.py_**.

As the describe of paper "_**VoxelMorph: A Learning Framework for Deformable Medical Image Registration**_", the loss function includes the _Lsim_, _Lsmooth_ and _Lseg_. _Lseg_ represents the anatomical segmentation information used during training which improves the effect of registration. 

I processed the OASIS-1 dataset by using "recon-all" command of freesurfer and preprocessed the dataset same as mentioned in the paper. But the homepage shows nothing about how to use label file when train the registration model, now I don't know how to import the segmentation information when training the model.

I guessed that "**_--labels_**" of generation parameters is the keyword of import segmentation information. I have tried some files to train the network: _aseg.nii.gz_, _aseg.stats_, _lh.aparc.annot_, _lh.cortex.label_, _aparc.annot.ctab_.  

The commands I used (e.g.): 
_python train_synth_reg.py --data-dir /public/home/jd_mxy/First_reg_OASIS --labels aseg.stats_
 (I haved appointed the location of _--model-dir_ in _train_synth_reg.py_, and I changed the _--epochs_ and _--save-freq_ to meet the need of mine).

But all these files return running error information:
![image](https://user-images.githubusercontent.com/52778658/113145825-81acca80-9261-11eb-81a9-7101b3a11316.png)

I tried to upgrade or downgrade the version of numpy, but it doesn't work. I don't know whether the question is caused by the version of numpy, or the file type of I input.

So, whether the keyword is _--labels_ or not? What's the file type of labels? From all the _recon-all_  FreeSurfer files, which one I can used as the anatomical segmentation information? 

Sorry for bother you for this, I carefully browsed the issue and discussion about label, but I didn’t get the answer I wanted.
Hope to receive your answer, thanks for your patient answer!

**Comment by adalca** [comment]
The training file you're looking for is `train_semisupervised_seg.py`.

Currently, the dataloader assumes that the volume and segmentation are both stored in the npz file --- so for each subject you have an npz file like `scan1.npz` which has two variables: `vol` and `seg`. 

You can modify the dataloader to use the aseg.nii.gz file, which is where your labels are.

I believe @ahoopes was working on enabling nifti files in the loader, but im not sure of the state of that.

**Comment by MoXiaoYing96** [comment]
> The training file you're looking for is `train_semisupervised_seg.py`.
> 
> Currently, the dataloader assumes that the volume and segmentation are both stored in the npz file --- so for each subject you have an npz file like `scan1.npz` which has two variables: `vol` and `seg`.
> 
> You can modify the dataloader to use the aseg.nii.gz file, which is where your labels are.
> 
> I believe @ahoopes was working on enabling nifti files in the loader, but im not sure of the state of that.


According your reply, maybe I need to use _train_semisupervised_seg.py_ to train a semi-supervised segmentation model, and then segment OASIS dataset again by using semi-supervised segmentation model, the image processed through this model owns two variables: _vol_ and _seg_, which can be used for _train_synth_reg.py training_, is that right?

I will carefully read the relevant content of _train_semisupervised_seg.py_, but I still wonder know how to use segmentation information in _train_synth_reg.py_, thanks for your detailed reply!

**Comment by adalca** [comment]
Hi @MoXiaoYing96 

I am a bit confused, I don't think you need `train_synth_reg.py` at all, this is from a different project called SynthMorph (https://arxiv.org/abs/2004.10282), and I don't believe this is what you're interested in

If you're just interested in semi-supervised training like the one in the TMI paper you mention, you should just use `train_semisupervised_seg`. This is not a segmentation model, this is a registration model that uses segmentations in a semi-supervised model, which I think is what you need, right?

**Comment by MoXiaoYing96** [comment]
> Hi @MoXiaoYing96
> 
> I am a bit confused, I don't think you need `train_synth_reg.py` at all, this is from a different project called SynthMorph (https://arxiv.org/abs/2004.10282), and I don't believe this is what you're interested in
> 
> If you're just interested in semi-supervised training like the one in the TMI paper you mention, you should just use `train_semisupervised_seg`. This is not a segmentation model, this is a registration model that uses segmentations in a semi-supervised model, which I think is what you need, right?
> 
> We should probably document this better :)

Your answer solved the problem that has troubled me for a long time. I did misunderstand the usage of train_synth_reg.py. I should use train_semisupervised_seg.py to train the registration network I want. Really appreciate your warm help!

**Comment by adalca** [comment]
Glad it helped!
